src/modules/zenoh/zenoh_config.cpp

```diff
@@ -166,47 +241,98 @@ int Zenoh_Config::cli(int argc, char *argv[])
 		}
 
 	} else if (argc == 4) {
-		if (strcmp(argv[1], "addpublisher") == 0) {
-			if (AddPubSub(argv[2], argv[3], ZENOH_PUB_CONFIG_PATH) > 0) {
-				printf("Added %s %s to publishers\n", argv[2], argv[3]);
+		if (strcmp(argv[1], "net") == 0) {
+			SetNetworkConfig(argv[2], argv[3]);
 
-			} else {
-				printf("Could not add uORB %s -> %s to publishers\n",  argv[3], argv[2]);
-			}
+		} else if (strcmp(argv[1], "delete") == 0) {
+			if (strcmp(argv[2], "publisher") == 0) {
+				int res = DeletePubSub(argv[3], ZENOH_PUB_CONFIG_PATH);
 
-		} else if (strcmp(argv[1], "addsubscriber") == 0) {
-			if (AddPubSub(argv[2], argv[3], ZENOH_SUB_CONFIG_PATH) > 0) {
-				printf("Added %s -> uORB %s to subscribers\n", argv[2], argv[3]);
+				if (res < 0) {
+					printf("Could not delete publisher topic %s\n", argv[3]);
+				}
 
-			} else {
-				printf("Could not add %s -> uORB %s to subscribers\n",  argv[2], argv[3]);
+			} else if (strcmp(argv[2], "subscriber") == 0) {
+				int res = DeletePubSub(argv[3], ZENOH_SUB_CONFIG_PATH);
+
+				if (res < 0) {
+					printf("Could not delete subscriber topic %s\n", argv[3]);
+				}
 			}
+		}
 
-		} else if (strcmp(argv[1], "net") == 0) {
-			SetNetworkConfig(argv[2], argv[3]);
+	} else if (argc >= 5) {
+		if (strcmp(argv[1], "add") == 0) {
+			if (strcmp(argv[2], "publisher") == 0) {
+				int instance = 0;
+
+				if (argc == 6) {
+					if (sscanf(argv[5], "%d", &instance) != 1 || instance < 0) {
+						printf("Invalid instance %s (must be an integer, 0 for the default instance or a specific instance's index)\n",
+						       argv[5]);
+						return 0;
+					}
+				}
+
+				if (AddPubSub(argv[3], argv[4], instance, ZENOH_PUB_CONFIG_PATH) > 0) {
+					printf("Added %s %s to publishers (instance %d)\n", argv[3], argv[4], instance);
+
+				} else {
+					printf("Could not add uORB %s:%d -> %s to publishers\n",  argv[3], instance, argv[4]);
+				}
+
+			} else if (strcmp(argv[2], "subscriber") == 0) {
+				int instance = 0;
+
+				if (argc == 6) {
+					if (sscanf(argv[5], "%d", &instance) != 1 || instance == 0 || instance == -1) {
+						printf("Invalid instance %s (must be an integer, 0 for the default instance or -1 for a new uOrb instance)\n", argv[5]);
+						return 0;
+					}
+				}
+
+				if (AddPubSub(argv[3], argv[4], instance, ZENOH_SUB_CONFIG_PATH) > 0) {
+					printf("Added %s -> uORB %s:%d to subscribers\n", argv[3], argv[4], instance);
+
+				} else {
+					printf("Could not add %s -> uORB %s to subscribers\n",  argv[3], argv[4]);
+				}
+			}
```

Comment:
Maybe add an else here if neither `publisher` nor `subscriber` got matched and give some user feedback that there's a typo.